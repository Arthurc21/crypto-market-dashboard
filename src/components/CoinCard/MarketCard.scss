.market-card-contianer {
    display: flex;
    flex-direction: column;
    gap: 0.25rem;
    width: auto;
    height: 100%;
    background-color: transparent;
    color: #333333;
    width: 280px;
    margin: 2rem;

    .price-list {
        display: flex;
        flex-direction: column;
        gap: 0.125rem;

        .element-list {
            background-color: #f2f2f2;
            display: flex;
            justify-content: space-between;
            flex-direction: row;
        }
    }

    .card-header {
        background-color: #f2f2f2;
        .current-price-container {
            font-size: 1.25rem;
            letter-spacing: 0.125rem;
            font-weight: 600;
        }
    }
}