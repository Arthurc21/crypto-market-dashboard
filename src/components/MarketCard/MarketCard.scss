.market-card-contianer {
	display: flex;
	flex-direction: column;
	gap: 0.25rem;
	width: auto;
	height: 100%;
	background-color: transparent;
	color: #333333;
	width: 300px;
	margin: 1.25rem;

	.price-list {
		display: flex;
		flex-direction: column;
		gap: 0.125rem;

		.element-list {
			background-color: #f2f2f2; //f2f2f2
			display: flex;
			justify-content: space-between;
			flex-direction: row;
			line-height: 24px;
			padding: 0 8px;
			//border: 1px solid #777777;
		}
	}

	.card-header {
		display: flex;
		flex-direction: column;
		background-color: #f2f2f2;
		align-items: center;

		.current-price-container {
			height: 32px;
			display: flex;
			font-size: 1.25rem;
			letter-spacing: 0.125rem;
			font-weight: 600;
			align-items: center;
			justify-content: center;
		}

		.market-span {
			font-weight: 500;
			letter-spacing: 0.0625rem;
		}
	}
}
