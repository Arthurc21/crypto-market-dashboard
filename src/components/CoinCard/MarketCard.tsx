import React, { useEffect } from 'react'
import { TickerData } from '../../App'
import { Trade } from '../../types/Trade'
import './MarketCard.scss'

interface MarketCardProps {
	market: string
	lastPrices?: Array<TickerData>
}

export const MarketCard = ({ market, lastPrices }: MarketCardProps): React.ReactElement => {
	const currentPrice = lastPrices ? `$${lastPrices[0].price}` : ''
	const historicalPrices = lastPrices !== undefined ? lastPrices.slice(1, lastPrices.length) : []

    const formatDate = (timestamp: number) => {
        const fullDate = new Date(timestamp*1000).toISOString()
        const date = fullDate.slice(0, 10)
        const time = fullDate.slice(11, 19)
        return `${date} ${time}`
    }

	return (
		<div className="market-card-contianer">
			<div className="card-header">
				<div className="current-price-container">
					<span>{currentPrice}</span>
				</div>
				<span className='market-span'>{market.toUpperCase()}</span>
			</div>
			<div className="price-list">
				{historicalPrices.map((element, index) => {
					console.log('element', element)
					return (
						<div key={`price-${index}`} className="element-list">
							<span>{element.lastUpdate ? formatDate(element.lastUpdate) : ''}</span>
							<span>{element.price}</span>
						</div>
					)
				})}
			</div>
		</div>
	)
}
