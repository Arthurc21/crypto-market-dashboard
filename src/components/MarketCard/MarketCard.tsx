import React, { useEffect } from 'react'
import { TickerData } from '../../types/TickerData'
import './MarketCard.scss'
import { TriangleUpIcon, TriangleDownIcon, DashIcon } from '@primer/octicons-react'

interface MarketCardProps {
	market: string
	lastPrices?: Array<TickerData>
}

export const MarketCard = ({ market, lastPrices }: MarketCardProps): React.ReactElement => {
	const historicalPrices = lastPrices !== undefined ? lastPrices.slice(1, lastPrices.length) : []
	const redColor = '#B80000'
	const greenColor = '#8BC34A'
	const yellowColor = '#FFD54F'

	const formattPrice = new Intl.NumberFormat('en-US', {
		style: 'currency',
		currency: 'USD'
	})

	const formatDate = (timestamp: number) => {
		const fullDate = new Date(timestamp).toISOString()
		const date = fullDate.slice(0, 10)
		const time = fullDate.slice(11, 19)
		return `${date} ${time}`
	}

	const formatPriceValue = (): string => {
		let coinValue
		if (lastPrices && lastPrices[0].price !== undefined) {
			coinValue = lastPrices[0].price
			return formattPrice.format(coinValue)
		} else return '0'
	}

	const getStatusIcon = () => {
		if (lastPrices) {
			const currentPrice = lastPrices[0].price
			const prevPrice = lastPrices[1] ? lastPrices[1].price : 0
			if (currentPrice && prevPrice) {
				if (currentPrice > prevPrice) {
					return <TriangleUpIcon fill={greenColor} size="medium" />
				} else if (currentPrice < prevPrice) {
					return <TriangleDownIcon fill={redColor} size="medium" />
				} else if (currentPrice === prevPrice) {
					return <DashIcon fill={yellowColor} size="medium" />
				}
			}
		}
	}

	return (
		<div className="market-card-contianer">
			<div className="card-header">
				<div className="current-price-container">
					{getStatusIcon()}
					<span>{(lastPrices && lastPrices[0].price !== undefined) ? formatPriceValue() : 'Loading'}</span>
				</div>
				<span className="market-span">{market ? market.toUpperCase() : ''}</span>
			</div>
			<div className="price-list">
				{historicalPrices.map((element, index) => {
					//console.log('element', element)
					return (
						<div key={`price-${index}`} className="element-list">
							<span>{element.lastUpdate ? formatDate(element.lastUpdate) : ''}</span>
							<span>{element.price ? formattPrice.format(element.price) : ''}</span>
						</div>
					)
				})}
			</div>
		</div>
	)
}
