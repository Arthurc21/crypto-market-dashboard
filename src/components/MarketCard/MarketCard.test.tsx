import { create } from 'react-test-renderer'
import { MarketCard } from './MarketCard'

describe('Input Testing', () => {
	test('input snapshot', () => {
		let market = create(
			<MarketCard
				market={'marketText'}
			/>
		)
		expect(market.toJSON()).toMatchSnapshot()
	})
})