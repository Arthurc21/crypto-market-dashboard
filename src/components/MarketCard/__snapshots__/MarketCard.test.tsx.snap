// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Testing input snapshot 1`] = `
<div
  className="market-card-contianer"
>
  <div
    className="card-header"
  >
    <div
      className="current-price-container"
    >
      <span>
        0
      </span>
    </div>
    <span
      className="market-span"
    >
      MARKETTEXT
    </span>
  </div>
  <div
    className="price-list"
  />
</div>
`;
